psychopy/demos/coder/basic/shutdown_key.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+
+"""
+This demo shows how to use a "shutdown key" to quit a running
+experiment. The key does not have to be checked manually, but
+rather gets handled automatically by PsychoPy.
+
+This works when calling win.flip() repeatedly (as in this example)
+and when using core.wait() (not demonstrated here). If using wait(),
+be sure to set the `hogCPUperiod` parameter equal to the entire
+waiting duration.
+"""
+
+from psychopy.visual import Window
+from psychopy.clock import CountdownTimer
+from psychopy.visual.text import TextStim
+from psychopy.preferences import prefs
+
+prefs.general['shutdownKeys'] = ['escape', 'f9']
+
+win = Window()
+timer = CountdownTimer()
+text_timer = TextStim(win, pos=(0, 0.2))
+text_esc = TextStim(
+    win,
+    text='Press ESC or F9 to quit prematurely.',
+    pos=(0, -0.2))
+
+timer.reset(10)
+while timer.getTime() > 0:
+    text_timer.text = 'Auto-quitting in %i sec.' % round(timer.getTime())
+    text_timer.draw()
+    text_esc.draw()
+    win.flip()
```

Comment:
I wonder if this whole demo could be made shorter to highlight the critical aspect. I'm thinking something like this (untested)

```python
    from psychopy import visual, preferences
    preferences.prefs.general['shudownKeys'] = ['escape', 'f9']  # set quit keys here.

    win = visual.Window()
    stim = visual.TextStim(win)
    for frame in range(600):  # 10 seconds on a 60 Hz monitor
        stim.text = u'Displaying frame %i of 600. Press %s to quit prematurely: %s' %(frame, ' or '.join(prefs.general['shutdownKeys']))
        stim.draw()
        win.flip()
```